components/ILIAS/Contact/classes/class.ilMailSearchObjectGUI.php

```diff
@@ -136,137 +137,48 @@ protected function addPermission(array $a_obj_ids): void
         $this->ctrl->redirectByClass(ilWorkspaceAccessGUI::class, 'share');
     }
 
-    protected function share(): void
+    private function shareObjects(): void
     {
-        $view = '';
-        if ($this->http->wrapper()->query()->has('view')) {
-            $view = $this->http->wrapper()->query()->retrieve('view', $this->refinery->kindlyTo()->string());
-        }
-
-        if ($view === 'myobjects') {
-            $obj_ids = [];
-            if ($this->http->wrapper()->query()->has('search_' . $this->getObjectType())) {
-                $obj_ids = [
-                    $this->http->wrapper()->query()->retrieve(
-                        'search_' . $this->getObjectType(),
-                        $this->refinery->kindlyTo()->int()
-                    )
-                ];
-            } elseif ($this->http->wrapper()->post()->has('search_' . $this->getObjectType())) {
-                $obj_ids = $this->http->wrapper()->post()->retrieve(
-                    'search_' . $this->getObjectType(),
-                    $this->refinery->kindlyTo()->listOf($this->refinery->kindlyTo()->int())
-                );
-            }
+        $obj_ids = $this->http->wrapper()->query()->has('contact_mailinglist_search_obj_ids')
+            ? $this->http->wrapper()->query()->retrieve(
+                'contact_mailinglist_search_obj_ids',
+                $this->refinery->kindlyTo()->dictOf($this->refinery->kindlyTo()->int())
```

Comment:
This is not a `dictOf`, but a `listOf`. Please also try to use `byTring` and `always` to avoid the `has` check.